File: Listas_JP/Listas_ejemplo/src/LinkedList.java
import java.util.ArrayList;
import java.util.Collection;
import java.util.Collections;

public class LinkedList {
  Nodo PTR;
  Nodo ULT;

  public LinkedList(){
      this.PTR = null;
      this.ULT = null;
  }

  public void AddNode(int data){
     Nodo P = new Nodo(data);
     if (this.PTR == null){
          this.PTR = P;
          this.ULT = P;
      }else{
          this.ULT.next = P;
          this.ULT = P; 
      }
    this.ULT.next = this.PTR;
  }
  public void Escribir_lista(){
        Nodo P = this.PTR;
        System.out.print(P.data + "->");
        P = P.next;
        while(P != this.PTR){
            System.out.print(P.data +"->");
            P = P.next;
        }
        System.out.println(P.data);
    }

    public void Sort(){
        ArrayList<Integer> numeros = new ArrayList<>();
        Nodo P = this.PTR;
        numeros.add(P.data);
        P = P.next;
        while(P != this.PTR){
            numeros.add(P.data);
            P = P.next;
        }
        Collections.sort(numeros);
        P = this.PTR;
        int i = 0;
        P.data = numeros.get(i);
        P = P.next;
        while(P != this.PTR){
            i = i + 1;
            P.data = numeros.get(i);
            P = P.next;
        }
    }

}

File: Try_Lab2/LinkedList.py
from Nodo import Nodo

class LinkedList:
    def __init__(self):
        self.PTR = None
        self.ULT = None

    def AddNode(self,data):
        P = Nodo(data)
        if (self.PTR == None):
            self.PTR = P
            self.ULT = P 
        else:
            self.ULT.next = P
            self.ULT = P  

    def DeleteNode(self, data):
        if (self.PTR == None):
            print("Empty list")
        anteP = None
        P = self.PTR
        while(P != None):
            if (self.PTR.data == data):
                self.PTR = P.next
                del P
                P = self.PTR
            else:
                while(P != None and P.data != data):
                    anteP = P
                    P = P.next
                if (P != None and P.data == data):
                    anteP.next = P.next
                    P.next = None
                    del P
                    P = anteP.next
                else:
                    print("dato no en lista") 


            

    def Recorrido(self):
        P = self.PTR
        while(P != None):
            print(P.data, end="->")
            P = P.next
        print("None")
    
    def __repr__(self):
        respuesta = ""
        P = self.PTR
        while(P != None):
            respuesta = respuesta + str(P.Barrio.nombre) + "->"
            P = P.next
        respuesta = respuesta + "None"
        return respuesta
